### PR TITLE
Various MinGW build fixes

### PR DESCRIPTION
**Pull request by koprok**
- Fix vcpkg_fixup_cmake_targets() to work for MinGW
- Turn off missing import libs check for MinGW as it does not work
- Fix MinGW build of following ports: openal-soft, glew, protobuf

### CODE REVIEW

File: scripts/cmake/vcpkg_common_definitions.cmake [hunk 1]
```diff
@@ -111,3 +111,8 @@ set(CMAKE_IMPORT_LIBRARY_PREFIX "${VCPKG_TARGET_IMPORT_LIBRARY_SUFFIX}")
 
 set(CMAKE_FIND_LIBRARY_SUFFIXES "${VCPKG_FIND_LIBRARY_SUFFIXES}" CACHE INTERNAL "") # Required by find_library
 set(CMAKE_FIND_LIBRARY_PREFIXES "${VCPKG_FIND_LIBRARY_PREFIXES}" CACHE INTERNAL "") # Required by find_library
+
+# Missing import libs check does not work with MinGW.
+if(VCPKG_TARGET_IS_MINGW)
+    set(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
+endif()
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_common_definitions.cmake:118]: That belongs in the triplet

File: scripts/cmake/vcpkg_fixup_cmake_targets.cmake [hunk 2]
```diff
@@ -48,7 +48,7 @@ function(vcpkg_fixup_cmake_targets)
         set(_vfct_TARGET_PATH share/${PORT})
     endif()
 
-    if(NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
+    if(VCPKG_TARGET_IS_WINDOWS)
         set(EXECUTABLE_SUFFIX "\\.exe")
     else()
         set(EXECUTABLE_SUFFIX)
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_fixup_cmake_targets.cmake:54]: replace with `string(REPLACE "." "\\." EXECUTABLE_SUFFIX "${VCPKG_TARGET_EXECUTABLE_SUFFIX}")`

**Comment by JackBoosY** [comment]
Hmm...I don't know how many regressions are realted.

**Comment by koprok** [comment]
> Hmm...I don't know how many regressions are realted.

@JackBoosY, I don't think the regressions are related to my changes. The only change that affects all ports is the one in vcpkg_fixup_cmake_targets and it's really trivial.

What do you suggest me to do?

I could split the port and triplet changes in a separate merge request. They won't cause rebuild of all ports and so no (detected) regressions? Do you want me to do it this way, so that the port and triplet changes can be merged easier / faster?

**Comment by JackBoosY** [comment]
@koprok We are fixing CI errors now, please wait.

**Comment by JackBoosY** [comment]
- quickfix:arm64-windows needs retrigger test
 - rapidxml:arm-uwp needs retrigger test
 - podofo:arm-uwp needs retrigger test
 - mp3lame:arm-uwp needs retrigger test
 - sdl2:arm-uwp needs retrigger test
 - sigslot:arm-uwp needs retrigger test
 - cunit:arm-uwp needs retrigger test
 - unixodbc:x64-linux #11613 
 - libsquish:x64-linux needs retrigger test
 - osg:x64-linux fixing
 - rapidxml:x64-uwp needs retrigger test
 - libnice:x64-windows needs retrigger test
 - plplot:x64-windows needs retrigger test
 - pdal:x64-windows needs retrigger test
 - gmp:x64-windows #11565
 - ocilib:x64-windows needs retrigger test
 - podofo:x64-windows-static needs retrigger test
 - pdal:x64-windows-static needs retrigger test
 - dcmtk:x64-windows-static needs retrigger test
 - portmidi:x64-windows-static needs retrigger test
 - gmp:x64-windows-static #11565
 - zxing-cpp:x64-windows-static #11586
 - libfabric:x64-windows-static needs retrigger test
 - irrlicht:x86-windows needs retrigger test
 - argtable2:x86-windows needs retrigger test
 - gsoap:x86-windows should be fixed
 - pdal:x86-windows needs retrigger test
 - gmp:x86-windows #11565
 - tinyfiledialogs:x86-windows needs retrigger test
 - ocilib:x86-windows needs retrigger test

**Comment by JackBoosY** [comment]
Okay, all regressions are tracked.

**Comment by koprok** [comment]
Hi guys! I am going to split this merge request in two separate merge requests:

- one that contains the changes to the ports only; I have also accumulated more port changes and I will add them too;
- another one that contains only the change in vcpkg_fixup_cmake_targets; this is what causes complete rebuild of all ports and many regressions not related to my changes.

This way the changes to the ports (which are the most of my changes) could be merged much easier without having to resolve all other (unrelated) ports regressions before that.

**Pull request closed without merging**